packages/compass-user-data/src/user-data.ts

```diff
@@ -333,3 +337,143 @@ export class FileUserData<T extends z.Schema> extends IUserData<T> {
     return await this.readOne(id);
   }
 }
+
+// TODO: update endpoints to reflect the merged api endpoints
+export class AtlasUserData<T extends z.Schema> extends IUserData<T> {
+  private atlasService: AtlasService;
+  private readonly preferences = preferencesLocator();
+  // private readonly preferences = null;
+  private readonly connectionInfoRef = connectionInfoRefLocator();
+  // private readonly connectionInfoRef = null;
+  private readonly logger = createLogger('ATLAS-USER-STORAGE');
+  private readonly groupId =
+    this.connectionInfoRef?.current?.atlasMetadata?.projectId;
+  private readonly orgId =
+    this.connectionInfoRef?.current?.atlasMetadata?.orgId;
+  // should this BASE_URL be a parameter passed to the constructor?
+  // this might make future usage of this code easier, if we want to call a different endpoint
+  private readonly BASE_URL = 'cluster-connection.cloud-local.mongodb.com';
+  constructor(
+    validator: T,
+    { serialize, deserialize }: AtlasUserDataOptions<z.input<T>>
+  ) {
+    super(validator, { serialize, deserialize });
+    const authService = atlasAuthServiceLocator();
+    // const authService = null;
+    const options = {};
+
+    this.atlasService = new AtlasService(
+      authService,
+      this.preferences,
+      this.logger,
+      options
+    );
```

Comment:
This should be pulling an existing instance of the service from the service injection, not creating a new version of it